Resources/Prototypes/Entities/Objects/Weapons/Guns/Ammunition/Magazines/rifle.yml

```diff
@@ -74,3 +74,57 @@
       map: ["enum.GunVisualLayers.Base"]
     - state: mag-1
       map: ["enum.GunVisualLayers.Mag"]
+
+- type: entity
+  id: MagazineRifleM41
+  name: "M41 magazine (10x24mm)"
+  parent: BaseMagazineRifle
+  components:
+  - type: BallisticAmmoProvider
+    whitelist:
+      tags:
+        - CartridgeRifle10x24mm
+    proto: CartridgeRifle10x24mm
+    capacity: 40
+  - type: Sprite
+    layers:
+    - state: base
+      map: ["enum.GunVisualLayers.Base"]
+    - state: mag-1
+      map: ["enum.GunVisualLayers.Mag"]
+
+- type: entity
+  id: MagazineRifleM41AE2
+  name: "M41AE2 magazine (10x24mm)"
+  parent: BaseMagazineRifle
+  components:
+  - type: BallisticAmmoProvider
+    whitelist:
+      tags:
+        - CartridgeRifle10x24mm
+    proto: CartridgeRifle10x24mm
+    capacity: 300
+  - type: Sprite
+    layers:
+    - state: base
+      map: ["enum.GunVisualLayers.Base"]
+    - state: mag-1
+      map: ["enum.GunVisualLayers.Mag"]
+
+- type: entity
+  id: MagazineRifleM4RA
+  name: "M4RA magazine (10x24mm)"
+  parent: BaseMagazineRifle
+  components:
+  - type: BallisticAmmoProvider
+    whitelist:
+      tags:
+        - CartridgeRifle10x24mm
+    proto: CartridgeRifle10x24mm
+    capacity: 25
+  - type: Sprite
+    layers:
+    - state: base
+      map: ["enum.GunVisualLayers.Base"]
+    - state: mag-1
+      map: ["enum.GunVisualLayers.Mag"]
```

Comment:
`- type: Sprite
    layers:
    - state: base
      map: ["enum.GunVisualLayers.Base"]
    - state: mag-1
      map: ["enum.GunVisualLayers.Mag"]`
      Seems to be used on all the mags, consider making a parent mag since everything here is the same anyway?

Reply:
I’m not that familiar with Yaml, so correct me if I’m wrong. I thought that the type above was to specify the sprite and needs the states in order to function correctly.

Reply:
Theoretically since they all use the same exact state and map names, it should just work.
Setting a entity parent will cause it to inherit everything from the parent

Reply:
Oh I see, it's because you haven't made the RSI for these files yet.